Sources/StreamChatUI/Utils/ImageLoading/NukeImageLoader.swift

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Nuke
+import UIKit
+
+extension ImageTask: Cancellable {}
+
+/// The class which is resposible for loading images from URLs.
+/// Internally uses `Nuke`'s shared object of `ImagePipeline` to load the image.
+open class NukeImageLoader: ImageLoading {
+    @discardableResult
+    open func loadImage(
+        using urlRequest: URLRequest,
+        cachingKey: String?,
+        completion: @escaping ((Result<UIImage, Error>) -> Void)
+    ) -> Cancellable? {
+        guard !SystemEnvironment.isTests else {
+            // When running tests, we load the images synchronously
+            let image = UIImage(data: try! Data(contentsOf: urlRequest.url!))!
+            completion(.success(image))
+            return nil
+        }
+
+        let request = ImageRequest(
+            urlRequest: urlRequest,
+            options: ImageRequestOptions(filteredURL: cachingKey)
+        )
+        
+        let imageTask = ImagePipeline.shared.loadImage(with: request) { result in
```

Comment:
`ImagePipeline.shared.loadImage(with: request)`
this is tightly coupling code, breaking D from SOLID, and makes test unreliable
should be exchanged with dependency injection of protocol usage